packages/cli-lib/api/hubs.js

```diff
@@ -0,0 +1,27 @@
+const request = require('request-promise-native');
+const { getRequestOptions } = require('../http/requestOptions');
+const { ENVIRONMENTS } = require('../lib/constants');
+const HUBS_API_PATH = 'sandbox-hubs/v1/self';
+
+async function fetchHubData(accessToken, portalId, env = ENVIRONMENTS.PROD) {
```

Comment:
Did you try using `https` for handling the request? I'm not sure if it would work for this use case or not, but it handles a lot of what you're manually doing here. Here's the file: https://github.com/HubSpot/hubspot-cli/blob/master/packages/cli-lib/http.js

Reply:
Yep I had tried using that file at first, but I kept running into an issue where it said something along the lines of "http is undefined/unavailable" when calling it from this `hubs.js` file. Had to import it directly here hence the similarities, I probably should've explained first